expression/date_arith.go

```diff
@@ -126,6 +146,13 @@ func (da *DateArith) evalArgs(ctx context.Context, args map[interface{}]interfac
 	if iVal == nil || err != nil {
 		return mysql.ZeroTimestamp, 0, 0, 0, 0, errors.Trace(err)
 	}
+	// handle adddate(expr,days) or subdate(expr,days) form
+	if da.Form == DateArithDaysForm {
```

Comment:
Seems u have already get iVal in line 146,  how about define iVal first, then use `da.Form == DateArithDaysForm` condition?